src/wayland/handlers/screencopy/mod.rs

```diff
@@ -375,26 +364,20 @@ fn constraints_for_toplevel(
     let size = surface.geometry().size.to_buffer(1, Transform::Normal);
     let wl_surface = surface.wl_surface()?;
 
-    let mut _kms_renderer = None;
-    let renderer = match backend {
-        BackendData::Kms(ref mut kms) => {
+    let mut renderer = backend
+        .offscreen_renderer(|kms| {
             let dma_node = with_renderer_surface_state(&wl_surface, |state| {
                 let buffer = state.buffer()?;
                 let dmabuf = get_dmabuf(&*buffer).ok()?;
                 dmabuf.node()
```

Comment:
Hmm, not your fault obviously, but it seems like we have a small inconsistency here.
The built-in screenshot code uses `advertised_node_for_client`, while here we ask the surface buffer for the node. Not sure whats better honestly.

Reply:
Yeah. I tried to keep the behavior unchanged here, but it's a little unclear if the nodes it is attempt to use everywhere are ideal.